### PR TITLE
Add support for pull_request_target webhook event

### LINKED ISSUE #1834: Add support for pull_request_target webhook event
GitHub API documentation:
https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target

As mentioned in #1619, this webhook event needs to be supported.

This would be a great PR for any new contributor to this repo or a new Go developer.
All contributions are greatly appreciated!

Feel free to volunteer for any issue and the issue can be assigned to you so that others don't attempt to duplicate the work.

Please check out our [CONTRIBUTING.md](https://github.com/google/go-github/blob/master/CONTRIBUTING.md) guide to get started.

Thank you!

### PR DESCRIPTION
**Pull request by ganboonhong**
Fixes #1834.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @ganboonhong !
LGTM.

Awaiting second LGTM before merging.

**Review by wesleimp — APPROVED** [review]
💙

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Pull request merged**
